### PR TITLE
Use scummvm-ios7-libs-v3

### PR DESCRIPTION
**Pull request by larsamannen**
The scummvm-ios7-libs-v3 packages the pre-built static libraries into xcframeworks, allowing to have arm64 libraries for both iPhoneOS and iPhoneSimulator.

The new scummvm-ios7-libs-v3 contains some new libraries compared to the v2-libs:
 - faad
 - gif
 - mpeg2
 - theoradec
 - vex

Update the instructions using the scummvm-ios7-libs-v3.

The scummvm-ios7-libs-v3 can be downloaded at https://github.com/larsamannen/scummvm-ios7-libs-v3/blob/main/scummvm-ios7-libs-v3.zip

### CODE REVIEW

File: doc/docportal/other_platforms/ios.rst [hunk 1]
```diff
@@ -47,11 +47,11 @@ Create a new directory called ``build`` at the same level as the repository you
 
     mkdir build
 
-Copy the contents of the ``scummvm-ios7-libs-v2`` folder you extracted earlier into the ``build`` directory, using either the Finder, or the command line as follows:
+Copy the contents of the ``scummvm-ios7-libs-v3`` folder you extracted earlier into the ``build`` directory, using either the Finder, or the command line as follows:
 
 .. code-block::
 
-    cp -r ~/Downloads/scummvm-ios7-libs-v2/* ~/build/
+    cp -r ~/Downloads/scummvm-ios7-libs-v3/* ~/build/
```
Comment (criezy) [inline doc/docportal/other_platforms/ios.rst:54]: The new package is extracted as `frameworks` for me and not `scummvm-ios7-libs-v3/frameworks`.
So the instruction becomes: `cp -r ~/Downloads/frameworks ~/build/`
Or maybe we should suggest `mv ~/Downloads/frameworks ~/build/`

File: devtools/create_project/xcode.cpp [hunk 2]
```diff
@@ -477,61 +486,62 @@ void XcodeProvider::setupFrameworksBuildPhase(const BuildSetup &setup) {
 
 	// Local libraries
 	if (CONTAINS_DEFINE(setup.defines, "USE_FAAD")) {
-		DEF_LOCALLIB_STATIC("libfaad");
+		DEF_LOCALXCFRAMEWORK("faad", projectOutputDirectory);
```
Comment (criezy) [inline devtools/create_project/xcode.cpp:489]: Removing those `DEF_LOCALIB_STATIC` breaks the macOS build. Only a few libraries remain in the link list (those that you did not change as they are not used on iOS):

image
This results in the link phase failing.

**Review by criezy — CHANGES_REQUESTED** [review]
The changes to create_project break the macOS build. So this needs to be fixed.

At this point I have no plan to switch to xcframeworks for macOS as the main way for developers to get their libs is through homebrew or macports. Also for the i386 and x86_64 targets I use old Xcode versions that contain the SDKs I target, and those do not support xcframeworks. So for macOS it still needs to use the static libraries.

I think that it could actually be a good idea for iOS to have a switch in create_project (for example `--use-ios-static-libs` so that those with older Xcode versions can still use the v2 package.

Another change that will be needed is to update the configure/make way to build scummvm so that it works with the xcframework when building for ios. From what I can see it should work with the new v3 package, but would need to add `<libname>.xcframeworks/ios-arm64-armv7-armv7s/Headers/`to the header path and `<libname>.xcframeworks/ios-arm64-armv7-armv7s/` to the link path.
Actually, maybe the `--use-ios-static-libs` in create_project could also use the v3 package in the same way so that even old Xcode version can use the updated package?

**Comment by criezy** [comment]
I forgot to confirm that the change work well for me for iOS development on a M1 mac (both simulator and real device).

**Comment by larsamannen** [comment]
@criezy Updated according to comments. Ready to merge if everything looks ok!

File: doc/docportal/other_platforms/ios.rst [hunk 3]
```diff
@@ -47,11 +47,11 @@ Create a new directory called ``build`` at the same level as the repository you
 
     mkdir build
 
-Copy the contents of the ``scummvm-ios7-libs-v2`` folder you extracted earlier into the ``build`` directory, using either the Finder, or the command line as follows:
+Move the contents of the ``scummvm-ios7-libs-v3`` folder you extracted earlier into the ``build`` directory, using either the Finder, or the command line as follows:
```
Comment (criezy) [inline doc/docportal/other_platforms/ios.rst:50]:
```suggestion
Move the ``frameworks`` folder you extracted earlier from the ``scummvm-ios7-libs-v3.zip`` file into the ``build`` directory, using either the Finder, or the command line as follows:
```

**Review by criezy — APPROVED** [review]
I only made a minor suggestion for the documentation, but it otherwise looks good now.

**Pull request merged**